tests/test_ai_sidecar_external.py

```
@@ -0,0 +1,204 @@
+- Action with unknown name surfaces as an ``error`` frame from the sidecar.
+
+Tests use plain ``asyncio.run`` — repo has no pytest-asyncio dependency.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import json
+import socket
+from contextlib import asynccontextmanager
+from typing import AsyncIterator
```

Ruff (0.15.10)

[warning] Import from `collections.abc` instead: `AsyncIterator`

Import from `collections.abc`

(UP035)